tests/unit/test_run.py

```diff
@@ -0,0 +1,114 @@
+import pipeline
+import pytest
+import requests
+
+from interactive import run
+from tests.factories import AnalysisRequestFactory
+
+
+@pytest.fixture
+def workspace_repo(tmp_path):
+    repo = tmp_path / "workspace_repo"
+    repo.mkdir()
+    # --bare means we can push to it, as it has no checkout
+    run.git("init", "--bare", cwd=repo)
+    return repo
+
+
+def test_write_files(tmp_path):
+    analysis_request = AnalysisRequestFactory()
+    checkout = tmp_path / "checkout"
+    checkout.mkdir()
+    run.write_files(checkout, analysis_request, "codelist")
+
+    project = checkout / "project.yaml"
```

Comment:
I've been trying to phase out the use of `project` for this type of work, opting for `pipeline_config` in this situation.  I can't do much about the name of the file but we've discussed renaming it at some point.  The pipeline library tries very hard to avoid using that name and my upcoming changes to job-runner to use that library do the same.  Since this is a new project (see!), could we avoid using it here too?  It avoids confusion with the other 2-3 things which are also projects on the platform.

Reply:
I'm mainly using pipeline to validate that the generated yaml is valid, and only in tests.